### PR TITLE
Update documentation for Quarkus distribution

### LINKED ISSUE #1388: Update documentation for Quarkus distribution
### Description

Update documentation for the switch to Quarkus distribution being the default distribution. WildFly steps are being left in the documentation for now until we remove the WildFly distribution, but won't be displayed in community documentation.

List of current documentation and what needs updating:

* Authorization services
* Securing applications
* Server admin
* Server development
* Upgrading

The following does not need updating:

* Server installation and configuration: replaced by new server guides (https://github.com/keycloak/keycloak/tree/main/docs/guides)
* Getting started: replaced by new getting started guides, which will be updated through a separate issue
* OpenShift: RH-SSO only guides, which won't be updated for now

This issue does not cover migration from WildFly dist to Quarkus dist, which will be covered by separate issues.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by stianst**
- Add variables for Keycloak context paths to use '/' on Quarkus and '/auth' on WildFly
- Updated to authorization docs for Quarkus dist
- Updated to server admin docs for Quarkus dist
- Updated to server developer docs for Quarkus dist
- Updated to upgrade docs for Quarkus dist

The server installation docs is not being updated, as these will be replaced with new guides. Getting started guides are already replaced by new guides, and won't be updated. OpenShift guides are RH-SSO only and won't be updated at this point.

Leaving this one as separate commits to simplify review.

Closes #1388

### CODE REVIEW

File: server_admin/topics/authentication/x509.adoc [hunk 1]
```diff
@@ -76,6 +76,7 @@ The certificate identity mapping can map the extracted user identity to an exist
 
 The following sections describe how to configure {appserver_name}/Undertow and the {project_name} Server to enable X.509 client certificate authentication.
 
+ifeval::["{kc_dist}" == "wildfly"]
```
Comment (pedroigor) [inline server_admin/topics/authentication/x509.adoc]: We have `https-client-auth`. Do we need an example for Quarkus or that is already covered by TLS guide?

File: server_admin/topics/authentication/x509.adoc [hunk 2]
```diff
@@ -378,6 +380,7 @@ All certificates in trusted-ca-list-for-client-auth.pem must be added to link:{i
 
 {project_name} does not have built-in support for other reverse proxy implementations. However, you can make other reverse proxies behave in a similar way to `apache` or `haproxy`. If none of these work, create your implementation of the `org.keycloak.services.x509.X509ClientCertificateLookupFactory` and `org.keycloak.services.x509.X509ClientCertificateLookup` providers. See the link:{developerguide_link}[{developerguide_name}] for details on how to add your provider.
 
+ifeval::["{kc_dist}" == "wildfly"]
```
Comment (pedroigor) [inline server_admin/topics/authentication/x509.adoc:383]: Except for the `Dumping Headers` section, it should be possible to enable logging and follow the troubleshooting steps.

File: server_admin/topics/events/login.adoc [hunk 3]
```diff
@@ -126,6 +126,21 @@ The Logging Event Listener logs events to the `org.keycloak.events` log category
...
 
+ifeval::["{kc_dist}" == "quarkus"]
+. Change the log level for the `org.keycloak.events` category
+. Change the log level used by the Logging Event listener.
+
+To change the log level used by the Logging Event listener, add the following:
+
+[source,bash]
+----
+bin/kc.[sh|bat] start --spi-events-listener-success-level=info --spi-events-listener-error-level=error
```
Comment (pedroigor) [inline server_admin/topics/events/login.adoc]: It should be `--spi-events-listener-jboss-logging-success-level=info`. There is no default provider for the SPI so we need to reference the provider id.

File: server_development/topics/auth-spi.adoc [hunk 4]
```diff
@@ -849,7 +849,13 @@ org.keycloak.examples.authenticator.SecretQuestionRequiredActionFactory
 
 This services/ file is used by {project_name} to scan the providers it has to load into the system.
 
-To deploy this jar, just copy it to the `standalone/deployments` directory.
+ifeval::["{kc_dist}" == "quarkus"]
+To deploy this jar, just copy it to the `providers/` directory.
```
Comment (pedroigor) [inline server_development/topics/auth-spi.adoc]: Shall we mention the `build` step?

File: server_development/topics/providers.adoc [hunk 5]
```diff
@@ -466,7 +513,13 @@ The name of the script file. This property is *mandatory* and should map to a fi
 
 ==== Deploy the script JAR
 
+ifeval::["{kc_dist}" == "quarkus"]
+Once you have a JAR file with a descriptor and the scripts you want to deploy, you just need to copy the JAR to the {project_name} `providers/` directory.
```
Comment (pedroigor) [inline server_development/topics/providers.adoc]: And run the `build` command.

File: upgrading/topics/install_new_version-quarkus.adoc [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+It is important that you upgrade {project_name} server before upgrading the adapters.
+
+.Prerequisites
+* Handle any open transactions and delete the data/tx-object-store/ transaction directory.
+
+.Procedure
+. Download the new server archive
+. Move the downloaded archive to the desired location.
+. Extract the archive. This step installs a clean instance of the latest {project_name} release.
+. Copy `conf/`, `providers/` and `themes/` from the previous installation to the new installation.
```
Comment (pedroigor) [inline upgrading/topics/install_new_version-quarkus.adoc]: And make sure to run the `build` command to update the server configuration. Not sure if that is implicit though.

File: upgrading/topics/migrate_db-quarkus.adoc [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+For those cases, we added a threshold (the number of records) for automated index creation.
+By default, this threshold is `300000` records.
+When the number of records is higher than the threshold, the index is not created automatically,
+and there will be a warning message in server logs including SQL commands which can be applied later manually.
+
+To change the threshold, set the `index-creation-threshold` property, value for the default `connections-liquibase` provider:
+
+[source,bash]
+----
+kc.[sh|bat] start --spi-connections-liquibase-default-index-creation-threshold=300000
```
Comment (pedroigor) [inline upgrading/topics/migrate_db-quarkus.adoc:32]: Looks like the Quarkus implementation is missing this option. Created https://github.com/keycloak/keycloak/issues/9992.

**Comment by pedroigor** [comment]
@stianst I did not build the documentation yet. That is my next step as well as making sure I did not miss any of examples and missing instructions for Quarkus.

File: server_admin/topics/authentication/x509.adoc (hunk 1 above)
Comment (stianst) [reply server_admin/topics/authentication/x509.adoc:79]: That's all configured by the new guides, which is the correct place

File: server_admin/topics/authentication/x509.adoc (hunk 2 above)
Comment (stianst) [reply server_admin/topics/authentication/x509.adoc:383]: I think troubleshooting and stuff like that should be covered by the new guides, not in these old docs

File: server_admin/topics/authentication/x509.adoc (hunk 1 above)
Comment (stianst) [inline server_admin/topics/authentication/x509.adoc]:
```suggestion
ifeval::["{kc_dist}" == "wildfly"]
The following sections describe how to configure {appserver_name}/Undertow and the {project_name} Server to enable X.509 client certificate authentication.

```

Comment (stianst) [reply server_admin/topics/authentication/x509.adoc]: It should be covered in the new TLS guide, and not here. Looking at it now it's not explicitly mentioned, but "https-client-auth" is at least under the relevant options. I think that's sufficient for now, and we can extend on mtls coverage after the 17 release?

File: server_admin/topics/authentication/x509.adoc [hunk 8]
```diff
@@ -74,6 +74,7 @@ The certificate identity mapping can map the extracted user identity to an exist
 
 ==== Enable X.509 client certificate user authentication
 
+ifeval::["{kc_dist}" == "wildfly"]
 The following sections describe how to configure {appserver_name}/Undertow and the {project_name} Server to enable X.509 client certificate authentication.
```
Comment (stianst) [inline server_admin/topics/authentication/x509.adoc]:
```suggestion
ifeval::["{kc_dist}" == "wildfly"]
==== Enable X.509 client certificate user authentication

The following sections describe how to configure {appserver_name}/Undertow and the {project_name} Server to enable X.509 client certificate authentication.
```

File: server_admin/topics/authentication/x509.adoc [hunk 9]
```diff
@@ -197,7 +199,7 @@ image:images/x509-configuration.png[X509 Configuration]
...
 
 *Canonical DN representation enabled*::
-Defines whether to use canonical format to determine a distinguished name. The official link:https://docs.oracle.com/javase/8/docs/api/javax/security/auth/x500/X500Principal.html#getName-java.lang.String-[Java API documentation] describes the format. This option affects the two User Identity Sources _Match SubjectDN using regular expression_ and _Match IssuerDN using regular expression_ only. Enable this option when you set up a new {project_name} instance. Disable this option to retain backward compatibility with existing {project_name} instances.
+Defines whether to use canonical format to determine a distinguished name. The official link:https://docs.oracle.com/javase/8/docs/api/javax/securityauth/x500/X500Principal.html#getName-java.lang.String-[Java API documentation] describes the format. This option affects the two User Identity Sources _Match SubjectDN using regular expression_ and _Match IssuerDN using regular expression_ only. Enable this option when you set up a new {project_name} instance. Disable this option to retain backward compatibility with existing {project_name} instances.
```
Comment (stianst) [inline server_admin/topics/authentication/x509.adoc]:
```suggestion
Defines whether to use canonical format to determine a distinguished name. The official link:https://docs.oracle.com/javase/8/docs/api/javax/security/auth/x500/X500Principal.html#getName-java.lang.String-[Java API documentation] describes the format. This option affects the two User Identity Sources _Match SubjectDN using regular expression_ and _Match IssuerDN using regular expression_ only. Enable this option when you set up a new {project_name} instance. Disable this option to retain backward compatibility with existing {project_name} instances.
```

File: server_admin/topics/authentication/x509.adoc (hunk 2 above)
Comment (stianst) [reply server_admin/topics/authentication/x509.adoc:383]: We should rather add a troubleshooting guide in the new server guides, and link to that. I would consider that as a post-17 follow-up task though

File: server_admin/topics/events/login.adoc (hunk 3 above)
Comment (stianst) [inline server_admin/topics/events/login.adoc]:
```suggestion
bin/kc.[sh|bat] start --spi-events-listener-jboss-logging-success-level=info --spi-events-listener-jboss-logging-error-level=error
```

File: server_development/topics/auth-spi.adoc (hunk 4 above)
Comment (stianst) [inline server_development/topics/auth-spi.adoc]:
```suggestion
To deploy this jar, copy it to the `providers/` directory, then run `bin/kc.[sh|bat] build`.
```

File: server_development/topics/auth-spi.adoc [hunk 10]
```diff
@@ -1121,7 +1127,13 @@ org.keycloak.authentication.forms.RegistrationRecaptcha
...
```
Comment (stianst) [inline server_development/topics/auth-spi.adoc]:
```suggestion
To deploy this jar, copy it to the `providers/` directory, then run `bin/kc.[sh|bat] build`.
```

File: server_development/topics/providers.adoc (hunk 5 above)
Comment (stianst) [inline server_development/topics/providers.adoc]:
```suggestion
Once you have a JAR file with a descriptor and the scripts you want to deploy, you just need to copy the JAR to the {project_name} `providers/` directory, then run `bin/kc.[sh|bat] build`.
```

File: server_development/topics/user-storage/packaging.adoc [hunk 11]
```diff
@@ -1,14 +1,17 @@
...
 In order for {project_name} to recognize the provider, you need to add a file to the JAR: `META-INF/services/org.keycloak.storage.UserStorageProviderFactory`. This file must contain a line-separated list of fully qualified classnames of the `UserStorageProviderFactory` implementations:
 
 ----
 org.keycloak.examples.federation.properties.ClasspathPropertiesStorageFactory
 org.keycloak.examples.federation.properties.FilePropertiesStorageFactory
 ----
 
-{project_name} supports hot deployment of these provider JARs. You'll also see later in this chapter that you can package it within and as Jakarta EE components.
+ifeval::["{kc_dist}" == "quarkus"]
+To deploy this jar, just copy it to the `providers/` directory.
```
Comment (stianst) [inline server_development/topics/user-storage/packaging.adoc]:
```suggestion
To deploy this jar, copy it to the `providers/` directory, then run `bin/kc.[sh|bat] build`.
```

File: server_development/topics/user-storage/simple-example.adoc [hunk 12]
```diff
@@ -256,7 +267,13 @@ The class files for our provider implementation should be placed in a jar.  You
 org.keycloak.examples.federation.properties.FilePropertiesStorageFactory
 ----
 
-Once you create the jar you can deploy it using regular {appserver_name} means: copy the jar into the `standalone/deployments/` directory or using the JBoss CLI.
+ifeval::["{kc_dist}" == "quarkus"]
+To deploy this jar, just copy it to the `providers/` directory.
```
Comment (stianst) [inline server_development/topics/user-storage/simple-example.adoc]:
```suggestion
To deploy this jar, copy it to the `providers/` directory, then run `bin/kc.[sh|bat] build`.
```

File: upgrading/topics/install_new_version-quarkus.adoc (hunk 6 above)
Comment (stianst) [inline upgrading/topics/install_new_version-quarkus.adoc]:
```suggestion
. Copy `conf/`, `providers/` and `themes/` from the previous installation to the new installation.
. Re-build the server with `bin/kc.[sh|bat] build` unless you are using auto-build
```

File: server_admin/topics/authentication/x509.adoc [hunk 13]
```diff
@@ -72,6 +72,7 @@ The certificate identity mapping can map the extracted user identity to an exist
 * Certificate KeyUsage validation.
 * Certificate ExtendedKeyUsage validation.
 
+ifeval::["{kc_dist}" == "wildfly"]
```
Comment (stianst) [inline server_admin/topics/authentication/x509.adoc:75]: Follow-up for new TLS guide: https://github.com/keycloak/keycloak/issues/10041

**Review by pedroigor — APPROVED** [review]

**Pull request merged**